test/FlapperUniV2.t.sol

```diff
@@ -85,55 +89,62 @@ contract FlapperUniV2Test is DssTest {
 
     SplitterMock   public splitter;
     FlapperUniV2   public flapper;
-    FlapperUniV2   public linkFlapper;
+    FlapperUniV2   public imxFlapper;
     MockMedianizer public medianizer;
-    MockMedianizer public linkMedianizer;
+    MockMedianizer public imxMedianizer;
 
-    address     DAI_JOIN;
+    address     USDS_JOIN;
     address     SPOT;
-    address     DAI;
-    address     MKR;
+    address     USDS;
+    address     SKY;
     address     USDC;
-    address     LINK;
     address     PAUSE_PROXY;
     VatLike     vat;
     VowLike     vow;
+    address     UNIV2_USDS_IMX_PAIR;
 
     address constant LOG                 = 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F;
 
+    address constant IMX                 = 0xF57e7e7C23978C3cAEC3C3548E3D615c346e79fF; // Random token that orders after USDS
     address constant UNIV2_FACTORY       = 0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f;
-    address constant UNIV2_DAI_MKR_PAIR  = 0x517F9dD285e75b599234F7221227339478d0FcC8;
-    address constant UNIV2_LINK_DAI_PAIR = 0x6D4fd456eDecA58Cf53A8b586cd50754547DBDB2;
+    address constant UNIV2_SKY_USDS_PAIR = 0x2621CC0B3F3c079c1Db0E80794AA24976F0b9e3c;
 
     event Exec(uint256 lot, uint256 sell, uint256 buy, uint256 liquidity);
 
     function setUp() public {
         vm.createSelectFork(vm.envString("ETH_RPC_URL"));
 
-        DAI_JOIN      = ChainlogLike(LOG).getAddress("MCD_JOIN_DAI");
+        USDS_JOIN     = ChainlogLike(LOG).getAddress("USDS_JOIN");
         SPOT          = ChainlogLike(LOG).getAddress("MCD_SPOT");
-        DAI           = ChainlogLike(LOG).getAddress("MCD_DAI");
-        MKR           = ChainlogLike(LOG).getAddress("MCD_GOV");
+        USDS          = ChainlogLike(LOG).getAddress("USDS");
+        SKY           = ChainlogLike(LOG).getAddress("SKY");
         USDC          = ChainlogLike(LOG).getAddress("USDC");
-        LINK          = ChainlogLike(LOG).getAddress("LINK");
         PAUSE_PROXY   = ChainlogLike(LOG).getAddress("MCD_PAUSE_PROXY");
         vat           = VatLike(ChainlogLike(LOG).getAddress("MCD_VAT"));
         vow           = VowLike(ChainlogLike(LOG).getAddress("MCD_VOW"));
+
+        UNIV2_USDS_IMX_PAIR = Univ2FactoryLike(UNIV2_FACTORY).createPair(USDS, IMX);
```

Comment:
We could maybe add a check that the pair doesn't already exist to make sure that the pair creation won't start failing at some point in the future.

Reply:
And we could move the code adding the initial liquidity (L135-139) under the same `if`.

Reply:
Done.

Reply:
> And we could move the code adding the initial liquidity (L135-139) under the same `if`.

Not sure if you missed that comment above ^?

Reply:
Ah yeah, I think I didn't have the second comment in the page when doing (or maybe just missed it). However wondering if we want to do that as I believe function that adapts the price assumes to have certain volume and even if the pair exists doesn't mean that has it. But will double check as might be confused

Reply:
Ok yeah, whatever volume is fine, just 0 is the problem. Not sure what we prefer, as probably if created will receive some volume, but we might have the edge case that was created but nobody deposited and checking that seems overkilling.

Reply:
Ah yes you're right, better to leave it as you did I think. Resolving.